File: 2138_Divide_a_String_Into_Groups_of_Size_k.py
class Solution(object):
    def divideString(self, s, k, fill):
        """
        :type s: str
        :type k: int
        :type fill: str
        :rtype: List[str]
        """
        groups = []
        n = len(s)
        for i in range(0, n, k):
            current_group = s[i:i + k]
            if len(current_group) < k:
                current_group += fill * (k - len(current_group))
            groups.append(current_group)
        return groups

File: 2138_Divide_a_String_Into_Groups_of_Size_k.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public String[] divideString(String s, int k, char fill) {
        List<String> groups = new ArrayList<>();
        int n = s.length();
        for (int i = 0; i < n; i += k) {
            StringBuilder currentGroup = new StringBuilder();
            for (int j = 0; j < k; j++) {
                if (i + j < n) {
                    currentGroup.append(s.charAt(i + j));
                } else {
                    currentGroup.append(fill);
                }
            }
            groups.add(currentGroup.toString());
        }
        return groups.toArray(new String[0]);
    }
}
